narwhals/dataframe.py

```diff
@@ -216,6 +443,99 @@ def __getitem__(self, col_name: str) -> Series:
         return Series(self._dataframe[col_name])
 
     def to_dict(self, *, as_series: bool = True) -> dict[str, Any]:
+        r"""
+        Convert DataFrame to a dictionary mapping column name to values.
+
+        Arguments:
+            as_series: If set to true ``True`` values are Series, otherwise
+                        values are Any.
+
+        Examples:
+            >>> import polars as pl
+            >>> import narwhals as nw
+            >>> df_pl = pl.DataFrame(
+            ...     {
+            ...         "A": [1, 2, 3, 4, 5],
+            ...         "fruits": ["banana", "banana", "apple", "apple", "banana"],
+            ...         "B": [5, 4, 3, 2, 1],
+            ...         "cars": ["beetle", "audi", "beetle", "beetle", "beetle"],
+            ...         "optional": [28, 300, None, 2, -30],
+            ...     }
+            ... )
+            >>> df = nw.DataFrame(df_pl)
+            >>> df
+            ┌─────────────────────────────────────────────────┐
+            | Narwhals DataFrame                              |
+            | Use `narwhals.to_native()` to see native output |
+            └─────────────────────────────────────────────────┘
+            >>> nw.to_native(df)
+            shape: (5, 5)
+            ┌─────┬────────┬─────┬────────┬──────────┐
+            │ A   ┆ fruits ┆ B   ┆ cars   ┆ optional │
+            │ --- ┆ ---    ┆ --- ┆ ---    ┆ ---      │
+            │ i64 ┆ str    ┆ i64 ┆ str    ┆ i64      │
+            ╞═════╪════════╪═════╪════════╪══════════╡
+            │ 1   ┆ banana ┆ 5   ┆ beetle ┆ 28       │
+            │ 2   ┆ banana ┆ 4   ┆ audi   ┆ 300      │
+            │ 3   ┆ apple  ┆ 3   ┆ beetle ┆ null     │
+            │ 4   ┆ apple  ┆ 2   ┆ beetle ┆ 2        │
+            │ 5   ┆ banana ┆ 1   ┆ beetle ┆ -30      │
+            └─────┴────────┴─────┴────────┴──────────┘
+            >>> import pprint
+            >>> pprint.pprint(df.to_dict(as_series=False))
+            {'A': [1, 2, 3, 4, 5],
+             'B': [5, 4, 3, 2, 1],
+             'cars': ['beetle', 'audi', 'beetle', 'beetle', 'beetle'],
+             'fruits': ['banana', 'banana', 'apple', 'apple', 'banana'],
+             'optional': [28, 300, None, 2, -30]}
+            >>> p = pprint.pformat(df.to_dict(as_series=True)).replace('\t', '    ')
+            >>> print(p)
```

Comment:
maybe we can just skip this example and only show the `as_series=False` one?

Reply:
I change it to use `# doctest: +SKIP` so we can still have it in the docs.